objects/src/transaction/proven_tx.rs

```diff
@@ -23,7 +23,7 @@ pub struct ProvenTransaction {
     account_update: TxAccountUpdate,
 
     /// A list of nullifiers for all notes consumed by the transaction.
-    input_notes: InputNotes<Nullifier>,
+    input_notes: InputNotes,
```

Comment:
I don't think `InputNotes` (at least the way it is now) will work here because proven transactions shouldn't contain anything beyond `(nullifier, note_id_or_zero)` tuples for each input note.

A couple of options of addressing this (there may be more):
1. Make `InputNotes` generic but instead of `ToNullifier` have a different trait which reduces an input note to `(nullifier, note_id_or_zero)` tuples.
2. Replace `InputNotes` with a new type - e.g., something like `ProvenInputNotes` which would basically be a wrapper around `Vec<(NoteNullifier, Word)>`.

Reply:
So this datastructure is supposed to wrap the data that will be sent to the network? And when you say it won't work, you mean if we have the complete notes here we will leak information to the rollup?

Reply:
> And when you say it won't work, you mean if we have the complete notes here we will leak information to the rollup?

Yes, for private notes we'll expose exactly which note is being consumed. Another issue is that we'll be sending more data than is needed (though, this extra data would be relatively small compared to the included STARK proof).

> So this datastructure is supposed to wrap the data that will be sent to the network?

The main reason for "wrapping" (rather than just using `Vec<(NoteNullifier, Word)>`) is to keep input note commitment logic computations somewhat encapsulated.

Reply:
I tried the wrapper type, ended up duplicated all the code. Moving back to the trait implementation.